source/instructors/authoring/assessments/parameterized.rst

```diff
@@ -5,59 +5,50 @@
+The **Generate Saple Parameters** button on the right-hand side will run the code in the editor and diaply in the mall box below the button a sample set of generated parameters. If there is an error with the code, the box will turn red and the error message will be displayed.
+
+Creating and using parameters
+*****************************
+To create a parameter, store a value in the ``codio_parameters`` dictionary. You can then refer to the parameter throughout the assessment within double curly brackets (otherwise known as mustache templating).
 
    .. image:: /img/guides/setupparams.png
       :alt: Setting up parameters
 
-   .. image:: /img/guides/param_instructions.png
-      :alt: Parameters in instructions
-
-   .. image:: /img/guides/param_commands.png
-      :alt: Parameters in commands
-
-Parameter variables
-*******************
-
-We are not using global variables but explicitly storing parameter values in a dictionary. Here is an example:
-
-.. code:: ini
-
-  import random
-  import uuid
-
-  # init codio_parameters dictionary
-  codio_parameters = dict()
-
-  # generate random values
-  random_int = random.randint(0, 100)
-  random_string = str(uuid.uuid4())
-  random_boolean = random.choice([True, False])
-  random_float = float(random.randint(0, 1000)/100)
-  random_array = []
-  random_array.append(random.randint(0, 100))
-  random_array.append(str(uuid.uuid4()))
-  random_dict = dict()
-  random_dict["int"] = random.randint(0, 100)
-  random_dict["str"] = str(uuid.uuid4())
-  random_dict["boolean"] = random.choice([True, False])
-
-  # add random values to codio_parameters dictionary
-  codio_parameters["int"] = random_int
-  codio_parameters["str"] = random_string
-  codio_parameters["boolean"] = random_boolean
-  codio_parameters["float"] = random_float
-  codio_parameters["array"] = random_array
-  codio_parameters["dict"] = random_dict
\ No newline at end of file
+See the first image on this page for an example of creating parameters. Once created, you can then refer to parameters throughout the other fields in your assessment (e.g. instructions on the **General** tab, fields on the **Execution** tab, rationale on the **Grading** tab).
+
+   .. image:: /img/guides/param_instructions2.png
+      :alt: Parameters in question field
+
+   .. image:: /img/guides/param_execution.png
+      :alt: Parameters in fields of multiple choice assessment
+
+Accessing parameters in Auto-grading scripts
+********************************************
+When using parameters with assessments executing auto-grading scripts, such as with :ref:`Advanced Code Tests <advanced-code-test>`, you can access parameters from the ``CODIO_PARAMETERS`` environment variable.
```

Comment:
It would be good to have an in doc example of this but it could be done later.

Reply:
I will need time to build one -- no one seems to have a good one handy